treeherder/webapp/api/performance_data.py

```diff
@@ -882,6 +899,54 @@ def _get_signatures(self, repository_name, framework, interval, no_subtests):
         signatures = self._get_signatures_values(signatures)
         return signatures
 
+    def _create_graph_links(
+        self,
+        base_repo_name,
+        new_repo_name,
+        base_revision,
+        new_revision,
+        framework,
+        timeRange,
+        signature,
+    ):
+        params = {}
+        highlightedRevisionKey = 'highlightedRevisions'
+        seriesKey = 'series'
+        timeRangekey = 'timeRange'
+        seriesParams = None
+
+        if new_repo_name == base_repo_name:
+            # if repo for base and new are not the same then make diff
+            # series data one for each repo, else generate one
+            params['series'] = [new_repo_name, signature, 1, framework]
+        else:
+            params['series'] = [
+                [base_repo_name, signature, 1],
+                [new_repo_name, signature, 1, framework],
+            ]
+
+        params[highlightedRevisionKey] = [base_revision, new_revision]
+        params[timeRangekey] = timeRange
+
+        # if repos selected are not the same
+        if len(params['series']) == 2:
+            seriesParams.append((seriesKey, params[seriesKey][0]))
+            seriesParams.append((seriesKey, params[seriesKey][1]))
+
+        else:
+            seriesParams = {seriesKey: params[seriesKey]}
+
+        highlightedRevisionsParams = [
+            (highlightedRevisionKey, params[highlightedRevisionKey][0]),
+            (highlightedRevisionKey, params[highlightedRevisionKey][1]),
+        ]
```

Comment:
url encode accepts a tuple, this will help since we have duplicate query param for highlighted revision key, base first and then new second